tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1454,38 +1454,164 @@ Status Converter::RenameAndMarkOutputTensors(
+    if (trt_algorithm_id != null_tactic) {
+      allowed_algorithm_ids.insert(trt_algorithm_id);
+    }
+
+#if !IS_TRT_VERSION_GE(8, 0, 0, 0)
+    disallowed_algorithm_ids = {
+      // turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_medium_nhwc_gelu_tn_v1
+      -5927686925093575778,
+      //turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_interior_nhwc_gelu_tn_v1
+      -3848538574386518527,
+      //turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_small_nhwc_gelu_tn_v1
+      -959009792490796596
+    };
+#endif  // !IS_TRT_VERSION_GE(8, 0, 0, 0)
+
+    // Check whether allowed and disallowed tactics are disjointed.
+    if (allowed_algorithm_ids.size() > 0 && disallowed_algorithm_ids.size() > 0) {
+      std::vector<int64_t> intersection;
+      for (int64_t allowed : allowed_algorithm_ids) {
+        if (disallowed_algorithm_ids.count(allowed) > 0) {
+          intersection.emplace_back(allowed);
+        }
+      }
+      if (intersection.size() > 0) {
+        VLOG(1) << "Intersection of allowed and disallowed tactics found "
+                   "non-empty.";
+
+        VLOG(1) << "List of allowed tactics:";
+        VLOG(1) << absl::StrJoin(allowed_algorithm_ids, ", ");
 
+        VLOG(1) << "List of disallowed tactics:";
+        VLOG(1) << absl::StrJoin(disallowed_algorithm_ids, ", ");
+
+        VLOG(1) << "List of tactics common to both allowed and disallowed "
+                   "tactics:";
+        VLOG(1) << absl::StrJoin(intersection, ", ");
+
+        VLOG(1) << "Aborting.";
+        assert(0);
+      }
+    }
+  }
   // Returns value in [0, nbChoices] for a valid algorithm.
   int32_t selectAlgorithms(const nvinfer1::IAlgorithmContext& algoContext,
                            const nvinfer1::IAlgorithm* const* algoChoices,
-                           int32_t nbChoices,
-                           int32_t* selection) noexcept override {
+                           int32_t nbChoices, int32_t* selection) override {
+
+    int nbSelections = 0;
+
     // TensorRT always provides more than zero number of algorithms
     // in selectAlgorithms.
     assert(nbChoices > 0);
 
-    // making sure that the requested TRT algorithm ID doesn't go above the
-    // max value accepted.
-    selection[0] = std::min(algorithm_id_, nbChoices);
-    return 1;
+    absl::optional<int64_t> forced_algorithm_id  = absl::nullopt;
+
+    if (! allowed_algorithm_ids.empty()) {
+      assert(allowed_algorithm_ids.size() == 1);
+      forced_algorithm_id = *allowed_algorithm_ids.begin(0);
+      // Making sure that the requested TRT algorithm ID doesn't go above
+      // the max value accepted.
+      forced_algorithm_id = std::min(forced_algorithm_id.value(),
+                                     static_cast<int64_t>(nbChoices));
+
+      VLOG(1) << "Forcing TRT algorithm selection to: ID = "
+              << forced_algorithm_id.value();
+      selection[nbSelections++] = forced_algorithm_id.value();
```

Comment:
I am trying to outline my proposed solution.

First remove this line.

Reply:
If you remove this line, where do you actually force the algorithm ?

Reply:
line 1603 in the current code. here is too early to select, as you will still assert later on.

Reply:
Sounds good with me. Fixed

Reply:
Fixed